src/Microsoft.FeatureManagement/ISessionManager.cs

```diff
@@ -21,8 +21,7 @@ public interface ISessionManager
         /// Queries the session manager for the session's feature state, if any, for the given feature.
         /// </summary>
         /// <param name="featureName">The name of the feature.</param>
-        /// <param name="enabled">The state from the session, if any.</param>
-        /// <returns>True if the session manager was able to provide a state for the feature. False otherwise.</returns>
-        Task<bool> TryGetAsync(string featureName, out bool enabled);
+        /// <returns>A result indicating whether the session was able to provide a the state of the feature, and the state, if any.</returns>
+        Task<ReadSessionResult> TryGetAsync(string featureName);
```

Comment:
Should the API name just be `GetAsync` (without Try)?

Reply:
If we do nullable boolean I feel like the name should remain the same.

Reply:
Normally the return value of Try* is to indicate whether the operation is successful and the operation result is returned via an out parameter. This is why the old API makes sense. Now that we return the operation result directly, which just happens to be boolean, the `Try` could be confusing.

Reply:
The `Try` semantic isn't as well defined in async methods, so I agree this could be confusing. After more thought I feel there is no need to have `Try` in this method name anyways. I'll change it to `GetAsync`. Thanks.

Reply:
@zhenlan updated. It appears I had pushed it to the wrong branch earlier so the update didn't show up.